src/views/Login.js

```diff
@@ -1,104 +1,145 @@
 
-import { H1 } from 'indigo-react';
-import View from 'components/View';
-import Tabs from 'components/Tabs';
+import { H4, Grid } from 'indigo-react';
 
 import Ticket from './Login/Ticket';
 import Mnemonic from './Login/Mnemonic';
 import Advanced from './Login/Advanced';
+import Hardware from './Login/Hardware';
 
-import useLifecycle from 'lib/useLifecycle';
 import * as need from 'lib/need';
-import { ROUTE_NAMES } from 'lib/routeNames';
 
 import { useHistory } from 'store/history';
 import { useNetwork } from 'store/network';
 import { useWallet } from 'store/wallet';
 import { usePointCursor } from 'store/pointCursor';
 
-const TABS = {
-  TICKET: Symbol('TICKET'),
-  MNEMONIC: Symbol('MNEMONIC'),
-  ADVANCED: Symbol('ADVANCED'),
+import View from 'components/View';
+import Tabs from 'components/Tabs';
+import Crumbs from 'components/Crumbs';
+import Footer from 'components/Footer';
+import { ForwardButton, OfflineButton } from 'components/Buttons';
+
+const NAMES = {
+  TICKET: 'TICKET',
+  MNEMONIC: 'MNEMONIC',
+  HARDWARE: 'HARDWARE',
+  ADVANCED: 'ADVANCED',
 };
 
-const tabViews = {
-  [TABS.TICKET]: Ticket,
-  [TABS.MNEMONIC]: Mnemonic,
-  [TABS.ADVANCED]: Advanced,
+const VIEWS = {
+  [NAMES.TICKET]: Ticket,
+  [NAMES.MNEMONIC]: Mnemonic,
+  [NAMES.HARDWARE]: Hardware,
+  [NAMES.ADVANCED]: Advanced,
 };
 
-const tabOptions = [
-  { title: 'Ticket', value: TABS.TICKET },
-  { title: 'Mnemonic', value: TABS.MNEMONIC },
-  { title: 'Advanced', value: TABS.ADVANCED },
+const OPTIONS = [
+  { text: 'Master Ticket', value: NAMES.TICKET },
+  { text: 'Mnemonic', value: NAMES.MNEMONIC },
+  { text: 'Hardware', value: NAMES.HARDWARE },
+  { text: 'Advanced', value: NAMES.ADVANCED },
 ];
 
 export default function Login() {
   // globals
-  const history = useHistory();
+  const { push, replaceWith, names } = useHistory();
   const { contracts } = useNetwork();
-  const { wallet, resetWallet } = useWallet();
+  const { wallet } = useWallet();
   const { pointCursor, setPointCursor } = usePointCursor();
 
   // inputs
-  const [currentTab, setCurrentTab] = useState(TABS.TICKET);
+  const [currentTab, setCurrentTab] = useState(NAMES.TICKET);
 
-  // on-mount
-  useLifecycle(() => {
-    resetWallet();
-  });
+  const goToActivate = useCallback(
+    () => replaceWith([{ key: names.LANDING }, { key: names.ACTIVATE }]),
+    [replaceWith, names]
+  );
 
-  const doContinue = async () => {
+  const goToPoints = useCallback(() => {
+    push(names.POINTS);
+  }, [push, names]);
+
+  const goToPoint = useCallback(() => {
+    goToPoints();
+    push(names.POINT);
+  }, [goToPoints, push, names]);
+
+  const doContinue = useCallback(async () => {
     const _wallet = need.wallet(wallet);
     const _contracts = need.contracts(contracts);
 
     // if no point cursor set by login logic, try to deduce it
     let deduced = pointCursor;
-    if (Nothing.hasInstance(deduced)) {
+    if (Maybe.Nothing.hasInstance(deduced)) {
```

Comment:
I think @jtobin and I decided some time ago to just consistently use them outside of their `Maybe.` namespace, and I believe all newer code adheres to that?

Reply:
oh no, I've been doing literally the opposite this whole time 😓

I'll probably have to do a pass against `Maybe.` and convert all of those inputs across the project